lib/OpenQA/Worker/Cache.pm

```diff
@@ -176,6 +186,16 @@ sub get_asset {
             next;
         }
         $ret = download_asset($job->{id}, lc($asset_type), $asset, ($result->{etag}) ? $result->{etag} : undef);
+        if ($ret =~ /5[0-9]{2}/ && $n--) {
```

Comment:
Couldn't the path returned by `download_asset` contain a substring `5xx`? If plain numbers are safe perhaps anchoring the regex (`/^5[0-9]{2}$/`) would work.